tests/source/issue-4027.rs

```diff
@@ -0,0 +1,30 @@
+mod a;
```

Comment:
Love the thoroughness of the tests :heart: Would you mind either: 

* Creating a new file like `reorder_imports_comments.rs` with separated blocks (with their own comment headers) and the exact snippets from #4027 and #3720, and then any additional scenarios can go below
* Creating separate files for those two issues with the exact snippets provided, and then any additional scenarios can go in new files 

It's really helpful for long term maintenance to structure the tests in such a way that issue-specific fixes/tests can be tied back directly